editor/src/components/canvas/ui/floating-insert-menu.tsx

```diff
@@ -305,6 +310,27 @@ function getMenuTitle(insertMenuMode: 'closed' | 'insert' | 'convert' | 'wrap'):
 export var FloatingMenu = betterReactMemo('FloatingMenu', () => {
   const colorTheme = useColorTheme()
 
+  // This is a ref so that changing the highlighted element does not trigger a re-render loop
+  // This is FINE because we only use the value in callbacks
+  const activelySelectedInsertOptionRef = React.useRef<InsertMenuItem | null>(null)
+
+  const ariaLiveMessages = React.useMemo(
+    () => ({
+      onFocus: ({ focused }: { focused: InsertMenuItem }) => {
+        activelySelectedInsertOptionRef.current = focused
+      },
+    }),
+    [],
+  )
+
+  const [filterInputValue, setFilterInputValue] = React.useState('')
+  const onInputValueChange = React.useCallback((newValue, actionMeta: InputActionMeta) => {
+    // when the user "tabs out" to the checkboxes, prevent react-select from clearing the input text
```

Comment:
If the user decides to select something on the canvas or perform some other (non-keyboard) interaction that would close the context menu, will this result in us still retaining the previous input value?

Reply:
nope, there's two components, one is a generic holding container that renders FloatingPopup : null. so closing the popup actually unmounts the FloatingPopup component, erasing all the internal state we might have had